zeeguu_api/api/user_working_session.py

```diff
@@ -0,0 +1,173 @@
+import zeeguu
+from zeeguu.model import UserWorkingSession
+
+from .utils.route_wrappers import cross_domain, with_session
+from .utils.json_result import json_result
+import flask
+from . import api
+
+
+# ---------------------------------------------------------------------------
+@api.route("/working_sessions_by_user", methods=("GET",))
+# ---------------------------------------------------------------------------
+@cross_domain
+@with_session
+def working_sessions_by_user():
+    """
+        Returns user_working_sessions data based on the provided filters for the specified user
+
+        Parameters:
+            user_id
+            from_date   (using ISO-8601 format)
+            to_date     (using ISO-8601 format)
+            is_active
+
+        Example: post
+        http://localhost:9001/working_sessions_by_user?user_id=534&from_date=2017-02-24T00:00:00&to_date=2018-02-24T09:13:13&is_active=False&session=92752705
+       
+        return:
+        [[session1], [session2], ... [sessionn]]
+        where session
+        [session_id, start_time, article_id, duration (in miliseconds)]
+        Example:
+        [
+            [
+                147,
+                "2018-02-24T09:11:11",
+                null,
+                61
+            ],
+            ...
+        ]
+    """
+    user_id =  int(flask.request.args['user_id'])
```

Comment:
i wonder whether i'm missing something.

but if not, then the user_id does not have to be passed explicitly in the call. 
the user id can be extracted from the session id, because the session id is linked to a user id.

or maybe is this function thought to be used by a teacher to obtain info about a student?
in that case, we must add a check to make sure that the one who requests info is teacher for the cohort of the one about which the info is being requested. however, the dashboard-dev-team has already created functionality for this, so we should reuse it.